akka-persistence/src/main/scala/akka/persistence/PersistentView.scala

```diff
@@ -361,6 +361,10 @@ trait PersistentView extends Actor with Snapshotter with Stash with StashFactory
     override def recoveryRunning: Boolean = false
 
     override def stateReceive(receive: Receive, message: Any): Unit = message match {
+      case ReplayedMessage(p) ⇒
+        // we can get ReplayedMessage here if it was stashed by user during replay
+        // unwrap the payload
+        PersistentView.super.aroundReceive(receive, p.payload)
```

Comment:
yes, unwrapping is necessary if we support stashing, otherwise an explicit failure would be needed. It feels like a bit risky to stash replayed messages, but if users need it I trust your judgement to add support for it.